Model/Payment_Method.php

```diff
@@ -172,11 +172,36 @@ public function getConfigFields(): array
             ],
             'paymentAction' => [
                 'type'        => 'select',
-                'field'       => 'oxpayments__wdoxidee_transactiontype',
+                'field'       => 'oxpayments__wdoxidee_transactionaction',
                 'options'     => TransactionModel::getTranslatedActions(),
                 'title'       => Helper::translate('config_payment_action'),
                 'description' => Helper::translate('config_payment_action_desc'),
             ],
         ];
     }
+
+    /**
+     * Returns array of setting names which should not be part of output debug infos
+     *
+     * @return array
+     */
+    public function getProtectedFieldNames()
+    {
+        return ['httpUser', 'httpPassword', 'secret'];
+    }
+
+    /**
+     * Returns an array of fields to be displayed in support email
+     *
+     * @return array
+     */
+    public function getSupportConfigFields()
+    {
+        $aFieldsPublic = array_filter($this->getConfigFields(), function ($field, $key) {
+            unset($field); //avoid unused local variable warnings
```

Comment:
If a warning is displayed here it's clearly a PHPMD issue. I would use `@SuppressWarnings`.